Maskbook/Scene/App/RedPacket/ViewModel/LuckyDropViewModel.swift

```diff
@@ -0,0 +1,728 @@
+        let message = self.message.isEmpty ? defaultMessage : self.message
+        
+        // tokenType
+        let tokenType = token.isMainToken == true ? BigUInt(0) : BigUInt(1)
+        // tokenAddr
+        var tokenAddr: String = ""
+        if token.isMainToken == true {
+            let chainKey = maskUserDefaults.network.redPacketConstantKey
+            tokenAddr = nativeTokenAddress?[chainKey].string ?? "0x0000000000000000000000000000000000000000"
+        } else if let address = token.contractAddress {
+            tokenAddr = address
+        }
+        guard let tokenAddressETH = EthereumAddress(tokenAddr) else { return }
+        
+        // totalTokens
+        guard let total = Web3.Utils.parseToBigUInt(
+            self.totalQuantity.stringValue,
+            decimals: Int(token.decimal)
+        ) else {
+            return
+        }
+        
+        let senderName = profileNickName ?? personaManager.currentPersona.value?.identifier ?? "Unknown User"
+        
+        let param = HappyRedPacketV4.CreateRedPacketInput(
+            publicKey: publicKeyETH,
+            number: number,
+            ifrandom: isRandom,
+            duration: duration,
+            seed: seed.bytes,
+            message: message,
+            name: senderName,
+            tokenType: tokenType,
+            tokenAddr: tokenAddressETH,
+            totalTokens: total)
+        
+        Task {
+            let tx = await ABI.happyRedPacketV4.createRedPacket(
+                token: token,
+                gasFeeViewModel: gasFeeViewModel,
+                param: param,
+                password: privateKey.toHexString().addHexPrefix()
+            )
+            await MainActor.run {
+                // Show the loading animation and reset the `ComfirmButton`'s state.
+                buttonType = .requestAllowance
+                checkParam()
+                if let tx = tx {
+                    walletBottomViewModel.observeTransaction(txHash: tx)
+                }
+            }
+        }
+    }
+    
+    // MARK: - Private method
+    @MainActor
+    private func getAllowance() async -> BigUInt? {
+        guard let fromAddress = maskUserDefaults.defaultAccountAddress,
+              let originalOwner = EthereumAddress(fromAddress) else {
+                  return nil
+              }
+        guard let web3 = Web3ProviderFactory.provider else { return nil }
+        guard let contractAddressStr = token?.contractAddress,
+              let contractAddress = EthereumAddress(contractAddressStr) else {
+                  return nil
+              }
+        guard let luckyDropAddressStr = luckyDropAddressStr,
+              let luckyDropAddress = EthereumAddress(luckyDropAddressStr) else {
+                  return nil
+              }
+        let erc20 = ERC20(web3: web3, provider: web3.provider, address: contractAddress)
+        
+        return await withCheckedContinuation { continuation in
+            Task.detached {
```

Comment:
Don't mess `withCheckedContinuation` and `Task.detached` together.  If you need the task be excuted without block main queue, then just wrap the `erc20.getAllowance(_:,_:)` into a separate async function and you can use `@MaskGroupActor`.